Sources/Tentacle/Branch.swift

```diff
@@ -20,36 +17,42 @@ extension Repository {
     }
 }
 
-public struct Branch {
+public struct Branch: ResourceType {
+
+    public struct Commit: Decodable {
+        public let sha: SHA
+    }
 
     /// Name of the branch
     public let name: String
 
-    /// Sha of the commit the branch points to
-    public let sha: SHA
+    /// The commit the branch points to
+    public let commit: Commit
```

Comment:
I'm still confused about why this isn't just a `SHA`. 🙈

Reply:
This is what the payload returns: 

```json
[
  {
    "name": "master",
    "commit": {
      "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
      "url": "https://api.github.com/repos/octocat/Hello-World/commits/c5b97d5ae6c19d5c5df71a34c7fbeeda2479ccbc"
    },
    "protected": true,
    "protection_url": "https://api.github.com/repos/octocat/Hello-World/branches/master/protection"
  }
]
```

I can retrieve the `sha` directly with a custom `init`, is that what you have in mind? Maybe I'm just missing something  🤷‍♂️

Reply:
Ah, okay! I didn't realize there was another property returned on `commit`! 😅